### PR TITLE
Fixes + Improvements for the release of pyopenms 2.0

### PR DESCRIPTION
**Pull request by uweschmitt**
The major improvements are to fix the difficulties we had to release pyopenms 1.11 on Mac OS and to support the creation of Python wheels, see http://pythonwheels.com/

Beyond that I had to remove two Methods from `MassTrace.pxd` which are declared in OpenMS header files but not implemented (any more). Further I changed `setup.py` so that the pyopenms version is now always in sync with OpenMS and follows the same notation for the version number.

### CODE REVIEW

File: src/pyOpenMS/CMakeLists.txt [hunk 1]
```diff
@@ -341,13 +381,26 @@ add_custom_target(pyopenms_create_cpp
...
-	COMMAND ${PYTHON_EXECUTABLE} setup.py build_ext
-	COMMAND ${PYTHON_EXECUTABLE} setup.py bdist_egg
-	COMMAND ${PYTHON_EXECUTABLE} setup.py bdist --format=zip
-	COMMAND ${PYTHON_EXECUTABLE} setup.py build_ext --inplace
-	DEPENDS pyopenms_create_cpp
-	WORKING_DIRECTORY ${CMAKE_BINARY_DIR}/pyOpenMS )
+IF(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
+    add_custom_target(pyopenms
+        COMMAND ${PYTHON_EXECUTABLE} setup.py build_ext
+        COMMAND fix_pyopenms_dependencies_on_mac.sh
```
Comment (grosenberger) [inline src/pyOpenMS/CMakeLists.txt]: If this command is executed in the OpenMS base directory, it fails.

Comment (uweschmitt) [inline src/pyOpenMS/CMakeLists.txt]: what is the reason or error message ? It does not fail on my machine, but I have a separate folder for building, so I do not compile inplace.

Comment (grosenberger) [inline src/pyOpenMS/CMakeLists.txt]: ld: warning: directory not found for option '-L/pyopenms/bin/Release'
ld: warning: directory not found for option '-L/pyopenms/Release'
ld: warning: directory not found for option '-L/pyopenms/contrib/lib'
/bin/sh: fix_pyopenms_dependencies_on_mac.sh: command not found
make[3]: *** [src/pyOpenMS/CMakeFiles/pyopenms] Error 127

Comment (uweschmitt) [inline src/pyOpenMS/CMakeLists.txt]: strange.... `setup.py` is found but the bash script next to it is not found !? 
I will try to fix this.

**Comment by uweschmitt** [comment]
@grosenberger : after cleaning the full `pyOpenMS`  folder in my build directory I could reproduce your issue. The last commit fixed it on my machine. Can you again try it ?

**Comment by grosenberger** [comment]
Great, it works now.

**Comment by timosachsenberg** [comment]
Hi @uweschmitt. Do we need this for the OpenMS 2.0 release? If yes, can we confirm that it works on all supported platforms?

**Comment by uweschmitt** [comment]
@grosenberger it would be nice to support python wheels with pyopenms 2.0 and I'm quite shure that the changes should work on other platforms too.

**Comment by hroest** [comment]
I currently get build errors when building this on CentOS:

```
-- Found setuptools version 0.6. The version is to old (>= 12.0 is required)
CMake Error at src/pyOpenMS/CMakeLists.txt:217 (message):
  Please upgrade setuptools or disable pyOpenMS.


-- Configuring incomplete, errors occurred!

```

**Comment by hroest** [comment]
would there be a way around this so that building would work on mac and centos systems?

**Comment by hroest** [comment]
I also cannot install a newer setuptools as the old version is still installed
```
% pip install --install-option="--prefix=/opt/imsb/hroest/"  setuptools Requirement already satisfied (use --upgrade to upgrade): distribute in /usr/lib/python2.6/site-packages
Cleaning up...
```

and as I dont have sudo, `--upgrade` also fails

```
OSError: [Errno 13] Permission denied: '/usr/bin/easy_install'
```

so to me the only options are 

- find a way to distribute a new setuptools with pyOpenMS
- change this pull request to make it work on older setuptools
- keep it as floating patch for Mac OSX only

**Comment by hroest** [comment]
I will try with virtualenv and report back:

```

source /opt/imsb/hroest/venvs/pyopenms-env/bin/activate
pip install --upgrade setuptools
pip install --upgrade autowrap
pip install --upgrade nose
pip install --upgrade numpy
make pyopenms
```

**Comment by hroest** [comment]
This is the result, it ran almost through:

```
creating dist
creating 'dist/pyopenms-2.0.0-py2.6-linux-x86_64.egg' and adding 'build/bdist.linux-x86_64/egg' to it
removing 'build/bdist.linux-x86_64/egg' (and everything under it)
usage: setup.py [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
   or: setup.py --help [cmd1 cmd2 ...]
   or: setup.py --help-commands
   or: setup.py cmd --help

error: invalid command 'bdist_wheel'
make[3]: *** [src/pyOpenMS/CMakeFiles/pyopenms] Error 1
make[2]: *** [src/pyOpenMS/CMakeFiles/pyopenms.dir/all] Error 2
make[1]: *** [src/pyOpenMS/CMakeFiles/pyopenms.dir/rule] Error 2
make: *** [pyopenms] Error 2
```

**Comment by hroest** [comment]
try again with

```
source /opt/imsb/hroest/venvs/pyopenms-env/bin/activate
pip install --upgrade setuptools
pip install --upgrade autowrap
pip install --upgrade nose
pip install --upgrade numpy
pip install wheel
pip install -U pip
pip install -U setuptools
make pyopenms
```

**Comment by uweschmitt** [comment]
You can not update `setuptools` if you created your virtualenv with `--system-site-packages`.
Either you are allowed to `su` or you have to use a virtualenv without this flag.

**Comment by timosachsenberg** [comment]
Hi,
what is the state of this PR? At least it needs a rebase or merge against master I guess.

**Pull request merged**